include/asterisk/crypto.h

```diff
@@ -188,6 +197,144 @@ AST_OPTIONAL_API(int, ast_crypto_loaded, (void), { return 0; });
+
+/*!
+ * \brief Register a certificate extension to openssl
+ * \param oid  The OID of the extension
+ * \param short_name The short name of the extension
+ * \param long_name The long name of the extension
+ * \retval <0 Extension was not successfully added
+ * \retval >= NID of the added extension
+ */
+int ast_crypto_register_x509_extension(const char *oid,
+	const char *short_name, const char *long_name);
+
+/*!
+ * \brief Return the data from a specific extension in a cert
+ * \param cert The cert containing the extension
+ * \param nid The NID of the extension
+ *            (0 to search locally registered extensions by short_name)
+ * \param short_name The short name of the extension
+ *                   (only for locally registered extensions)
+ * \note Either nid or short_name may be supplied.  If both are,
+ * nid takes precedence.
+ * \note The extension nid may be any of the built-in values
+ * in openssl/obj_mac.h or a NID returned by
+ * ast_crypto_register_x509_extension().
+ * \returns The data for the extension or NULL if not found
+ * \warning Do NOT attempt to free the returned buffer.
+ */
+ASN1_OCTET_STRING *ast_crypto_get_cert_extension_data(X509 *cert, int nid,
+	const char *short_name);
+
+/*!
+ * \brief Load an X509 Cert from a file
+ * \param filename PEM file
+ * \returns X509* or NULL on error
+ */
+X509 *ast_crypto_load_cert_from_file(const char *filename);
+
+/*!
+ * \brief Load an X509 Cert from a NULL terminated buffer
+ * \param buffer containing the cert
+ * \param size size of the buffer.
+ *             May be -1 if the buffer is NULL terminated.
+ * \returns X509* or NULL on error
+ */
+X509 *ast_crypto_load_cert_from_memory(const char *buffer, size_t size);
+
+/*!
+ * \brief Retrieve RAW public key from cert
+ * \param cert The cert containing the extension
+ * \param raw_key Address of char * to place the raw key
+ * \note raw_key must be freed by the caller after use
+ * \retval <=0 An error has occurred
+ * \retval >0 Length of raw key
+ */
+int ast_crypto_get_raw_pubkey_from_cert(X509 *cert,
+	unsigned char **raw_key);
+
+/*!
+ * \brief Create an empty X509 store
+ * \returns X509_STORE* or NULL on error
+ */
+X509_STORE *ast_crypto_create_cert_store(void);
+
+/*!
+ * \brief Load an X509 Store with either certificates or CRLs
+ * \param store X509 Store to load
+ * \param file Certificate or CRL file to load or NULL
+ * \param path Path to directory with hashed certs or CRLs to load or NULL
+ * \note At least 1 file or path must be specified.
+ * \retval <= 0 failure
+ * \retval 0 success
+ */
+int ast_crypto_load_cert_store(X509_STORE *store, const char *file,
+	const char *path);
+
+/*!
+ * \brief Check if the reftime is within the cert's valid dates
+ * \param cert The cert to check
+ * \param reftime to use or 0 to use current time
+ * \retval 1 Cert is valid
+ * \retval 0 Cert is not valid
+ */
+int ast_crypto_is_cert_valid(X509*cert, time_t reftime);
```

Comment:
Maybe name this `ast_crypto_is_cert_time_valid(...)` or something? `..._is_cert_valid` might be confusing.

Reply:
yeah, went back and forth on this. i'll add "time" or "time_range" back in.